framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -274,12 +282,34 @@ protected TreeAnnotator createTreeAnnotator() {
      * {@link org.checkerframework.framework.type.typeannotator.ImplicitsTypeAnnotator}
      * that adds annotations to a type based on the content of the type itself.
      *
+     * Subclass may override this method.  The default type annotator is a {@link ListTypeAnnotator}
+     * of the following:
+     * <ol>
+     *   <li> {@link IrrelevantTypeAnnotator}: Adds top to types not listed in
+     *  the {@link RelevantJavaTypes} annotation on the checker
+     *   <li> {@link PropagationTypeAnnotator}: Propagates annotation onto wildcards
+     *   <li> {@link ImplicitsTypeAnnotator}: Adds annotations based on {@link ImplicitFor}
+     *  meta-annotations
+     * </ol>
+     *
      * @return a type annotator
      */
     protected TypeAnnotator createTypeAnnotator() {
+        List<TypeAnnotator> typeAnnotators = new ArrayList<>();
+        RelevantJavaTypes relevantJavaTypes =
+                checker.getClass().getAnnotation(RelevantJavaTypes.class);
+        if (relevantJavaTypes != null) {
+            Class<?>[] classes = relevantJavaTypes.value();
+            // Must be first in order to annotated all irrelevant types that are not explicilty
+            // annotated.
+            typeAnnotators.add(
+                    new IrrelevantTypeAnnotator(
+                            this, getQualifierHierarchy().getTopAnnotations(), classes));
+        }
+        typeAnnotators.add(new PropagationTypeAnnotator(this));
         implicitsTypeAnnotator = new ImplicitsTypeAnnotator(this);
-
-        return new ListTypeAnnotator(new PropagationTypeAnnotator(this), implicitsTypeAnnotator);
+        typeAnnotators.add(implicitsTypeAnnotator);
+        return new ListTypeAnnotator(typeAnnotators.toArray(new TypeAnnotator[0]));
```

Comment:
Just for code readability, I would add that second constructor that takes a list directly and doesn't convert back and forth from arrays.